public/app/plugins/datasource/tempo/traceql/TempoQueryBuilderOptions.tsx

```diff
@@ -31,162 +32,173 @@ const parseIntWithFallback = (val: string, fallback: number) => {
-          isOpen={isOpen}
-          onToggle={toggleOpen}
-        >
-          <EditorField
-            label="Step"
-            tooltip="Defines the step for metric queries. Use duration notation, for example 30s or 1m"
+export const TempoQueryBuilderOptions = React.memo<Props>(
+  ({ onChange, query, searchStreaming, metricsStreaming, app }) => {
+    const styles = useStyles2(getStyles);
+    const [isOpen, toggleOpen] = useToggle(false);
+    const isAlerting = app === CoreApp.UnifiedAlerting || app === CoreApp.CloudAlerting;
+
+    if (!query.hasOwnProperty('limit')) {
+      query.limit = DEFAULT_LIMIT;
+    }
+
+    if (!query.hasOwnProperty('tableType')) {
+      query.tableType = SearchTableType.Traces;
+    }
+
+    if (!query.hasOwnProperty('metricsQueryType')) {
+      query.metricsQueryType = MetricsQueryType.Range;
+    }
+
+    if (isAlerting && query.metricsQueryType === MetricsQueryType.Range) {
+      onChange({ ...query, step: '30s', metricsQueryType: MetricsQueryType.Instant });
+    }
+
+    const onLimitChange = (e: React.FormEvent<HTMLInputElement>) => {
+      onChange({ ...query, limit: parseIntWithFallback(e.currentTarget.value, DEFAULT_LIMIT) });
+    };
+    const onSpssChange = (e: React.FormEvent<HTMLInputElement>) => {
+      onChange({ ...query, spss: parseIntWithFallback(e.currentTarget.value, DEFAULT_SPSS) });
+    };
+    const onTableTypeChange = (val: SearchTableType) => {
+      onChange({ ...query, tableType: val });
+    };
+    const onMetricsQueryTypeChange = (val: MetricsQueryType) => {
+      onChange({ ...query, metricsQueryType: val });
+    };
+    const onStepChange = (e: React.FormEvent<HTMLInputElement>) => {
+      onChange({ ...query, step: e.currentTarget.value });
+    };
+
+    // There's a bug in Tempo which causes the exemplars param to be ignored. It's commented out for now.
+
+    // const onExemplarsChange = (e: React.FormEvent<HTMLInputElement>) => {
+    //   const exemplars = parseInt(e.currentTarget.value, 10);
+    //   if (!isNaN(exemplars) && exemplars >= 0) {
+    //     onChange({ ...query, exemplars });
+    //   } else {
+    //     onChange({ ...query, exemplars: undefined });
+    //   }
+    // };
+
+    const collapsedSearchOptions = [
+      `Limit: ${query.limit || DEFAULT_LIMIT}`,
+      `Spans Limit: ${query.spss || DEFAULT_SPSS}`,
+      `Table Format: ${query.tableType === SearchTableType.Traces ? 'Traces' : 'Spans'}`,
+      '|',
+      `Streaming: ${searchStreaming ? 'Enabled' : 'Disabled'}`,
+    ];
+
+    const collapsedMetricsOptions = [
+      `Step: ${query.step || 'auto'}`,
+      `Type: ${query.metricsQueryType === MetricsQueryType.Range ? 'Range' : 'Instant'}`,
+      '|',
+      `Streaming: ${metricsStreaming ? 'Enabled' : 'Disabled'}`,
+      // `Exemplars: ${query.exemplars !== undefined ? query.exemplars : 'auto'}`,
+    ];
+
+    return (
+      <EditorRow>
+        <div className={styles.options}>
+          {!isAlerting && (
```

Comment:
Hide or disable options if we are in alerting i.e. only allow instant queries with a predefined step.

Reply:
Would it make sense to remove `Search` and `Service Graph` query types?

Reply:
Yeah why not